### DISCUSSION TITLE
kustomize resources external git repo cache

### DISCUSSION THREAD

**Discussion by suppix** [argoproj/argo-cd#5421] [Q&A]
Hello. 

I want to keep some part of my configuration in another git repo . 

In my main.git repo I have something like this: 

kustomization.yaml
```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
- ../../base
- ssh://git@git.mycompany.com:2222/deploy/conf.git
- registry-credentials.sealed.yaml
- default-serviceaccount.yaml

patchesStrategicMerge:
- efs-provisioner.yaml
- config-server.yaml
- api.yaml
- frontend.yaml
```

But when I change conf.git repo, argocd doesn't see any changes. As I can see in the logs argocd sees that main.git doesn't have any changes and doesn't check conf.git 

Is it possible somehow to fix this ? or it's wrong idea to use external repo like this ?

**Comment by jessesuen** [comment]
>  or it's wrong idea to use external repo like this ?

Correct. This behavior is expected and is a pattern we discourage because the commit SHA can change meaning, when the remote changes, so things like rollback are not deterministic.

**Comment by suppix** [comment]
Hm. Can you please advice what is the best way to manage several conf files externally ? 
Create a separate application ?  I need to delegate access to some parts of application  among several teams.

**Comment by rp-moha** [comment]
@jessesuen - Is there any way to force refresh to sync change in base repo - when we use kustomization - and ArgoCD app is connected to overlay repo ? overlay git repo using submodule to connect to base git repo.

**Comment by rp-moha** [comment]
Oh I was missing on updating git submodule, that was needed explicitly.,. out of purview of ArgoCD certainly.
